### PR TITLE
rr20 financial checks, load data script

### PR DESCRIPTION
**Pull request by kengie**
New script `validation_tool/rr20_financials_check.py` contains the second batch of validation checks on the RR-20 form. This compliments the script `validation_tool/rr20_service_check.py` that was in the last PR.  Instructions for how to run it are at the top of the file.

Also, I modified and renamed the old script `blackcat_2022_ntdreports_toBQ.py` to what is now called `data_to_BQ.py` to make it more generalizable, and I then used this to load several new tables to BigQuery from BlackCat. All of these are tables that only need to be created once and so this script represents actions that will not be automated. Details of what was loaded and how are at the top of the file.

### CODE REVIEW

File: validation_tool/rr20_financials_check.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+import datetime
+import logging
+
+'''Script for checking RR-20 NTD report for Financial Data. 
+Grabs data from GCS buckets for "this year" and "last year". 
+Writes validated data into:
+- a folder called "gs://calitp-ntd-report-validation/validation_reports_2023"
+
+To run from command line with the default datasources, navigate to folder and type: 
+python rr20_financials_check.py'''
```
Comment (csuyat-dot) [inline validation_tool/rr20_financials_check.py:13]: Tried running `python rr20_financials_check.py` with @KatrinaMKaiser on my local jupyter notebook but got error, but other checks ran.

> google.api_core.exceptions.NotFound: 404 Not found: Table cal-itp-data-infra:blackcat_raw.2023_revenue_vehicle_inventory was not found in location us-west2

Screenshot 2023-10-27 110011

**Review by csuyat-dot — COMMENTED** [review]
Left comment on rr-20 financial check script.

File: validation_tool/rr20_financials_check.py (hunk 1 above)
Comment (kengie) [reply validation_tool/rr20_financials_check.py:13]: @csuyat-dot sorry the table was renamed, I just renamed the table in the script to match it now. Please pull the latest changes and retry.

Comment (csuyat-dot) [reply validation_tool/rr20_financials_check.py:13]: Hi @kengie! i pulled the latest changes this morning and tried to run the script again. Looks like i got a different error at the end this time. Thoughts?
`AttributeError: 'Workbook' object has no attribute 'add_format'`

image

Comment (kengie) [reply validation_tool/rr20_financials_check.py:13]: Hi @csuyat-dot , can you check whether you have the package installed in your computer? https://stackoverflow.com/questions/59794843/xlsxwriter-error-attributeerror-workbook-object-has-no-attribute-add-format

1st answer in the link above. If it is upgrade the package, comment in the 1st answer.

**Comment by csuyat-dot** [comment]
Ran `python a10_facilities_check.py` yesterday and got `keyError: 'Total Facilities'`. Ran `pip install -r requirements.txt` today, then ran a10 facilities check again, still got same `keyError: 'Total Facilities'`
a10_facilities_check key error

**Comment by kengie** [comment]
> Ran `python a10_facilities_check.py` yesterday and got `keyError: 'Total Facilities'`. Ran `pip install -r requirements.txt` today, then ran a10 facilities check again, still got same `keyError: 'Total Facilities'`
> a10_facilities_check key error

@csuyat-dot so yes, this script is the first one I wrote and looking back at the [instructions and what the default arguments are](https://github.com/cal-itp/ntd-modernization/blob/validator-rr20/validation_tool/a10_facilities_check.py#L7-L21):

```'''This file loads one dataset (A-10 form) that originates from Black Cat.
To run from command line, navigate to folder: 
* To run with the default datasources, type: python a10_facilities_check.py
* To specify data source file, type: python a10_facilities_check.py --a10_data <filepath> --a10_lastyr_data <filepath> 
                                
Performs 3 checks:
* Check 1. Check that sum of total facilities for each agency, across all modes, is a whole number.
* Check 2. Check that the sum of all total facilities is not zero.
* Check 3. Check whether total gen purpose facilities (all but heavy maintenance) is > 1. If so mark as "failure".

For more details on forms, SEE THIS NOTEBOOK: "2b_a10_validation_development.ipynb"
'''

def get_arguments():
    """Get the data as input arguments (for now)"""
    parser = ArgumentParser(description="VOMS inventory check")
    parser.add_argument('--a10_data', default="data/2021_a10_submitted_partialdata.csv")
    parser.add_argument('--a10_lastyr_data', default = "data/2020_a10_submitted_partialdata.csv")

    args = parser.parse_args()
    return args
```
One needs to have 2022 and 2023 data and look in the GCS bucket for the latest file. At the time of creation, I didn't to have 2022 data and so I tested it out on 2021 and 2020 data. At this time I still don't have 2022 data - so it can't really be run. If we *did* have it, we could use this command line argument: 
```python a10_facilities_check.py --a10_data "gs://calitp-ntd-report-validation/blackcat_ntd_reports_2023_raw/
NTD_Stations_and_Maintenace_Facilities_A10_2023_2023-11-07.xlsx" --a10_lastyr_data "<filepath here>"
```

I think it's most efficient *not* to rewrite this right now since we're on the verge of ingesting data from the API anyways. **BUT** for 2024, you will want to use the above argument, except that `--a10_data` should refer to the latest 2024 excel file and `-a10_lastyr_data` should refer to the last 2023 Excel file in the GCS buckets. In other words, you can't really test this (lack of "last year's data") but what I just wrote is instructions for next year (2024).

**Comment by kengie** [comment]
@csuyat-dot also if you get such a KeyError, it means that the column is called something else. Open the Excel file and check what the column is called, modify code if/as needed. The error probably occurred because whatever Excel file it was looking at did not have a column with that name.

**Comment by csuyat-dot** [comment]
@kengie Thank you for clarifying! If you can, please include your clarification of the a10 facilities check script in any upcoming handoff documents. Other than that, we are good to go with this PR.

**Comment by kengie** [comment]
> @kengie Thank you for clarifying! If you can, please include your clarification of the a10 facilities check script in any upcoming handoff documents. Other than that, we are good to go with this PR.

@csuyat-dot thank you! Would you mind approving the PR?

**Review by csuyat-dot — APPROVED** [review]
good to go.

**Pull request merged**